data-ingestion-service/src/main/java/gov/cdc/dataingestion/kafka/integration/service/KafkaConsumerService.java

```diff
@@ -193,6 +196,45 @@ public void handleMessageForRawElr(String message,
         validationHandler(message, hl7ValidationActivated, dataProcessingEnable);
     }
 
+
+    /**
+     * Receive XML converted HL7 and save it to NBS Interface
+     * Description: The ELR ingestion endpoint that we have right now is modified to accept both
+     * plain text HL7 as well as HL7s converted to XML already.
+     * */
+    @RetryableTopic(
+            attempts = "${kafka.consumer.max-retry}",
+            autoCreateTopics = "false",
+            dltStrategy = DltStrategy.FAIL_ON_ERROR,
+            retryTopicSuffix = "${kafka.retry.suffix}",
+            dltTopicSuffix = "${kafka.dlt.suffix}",
+            // retry topic name, such as topic-retry-1, topic-retry-2, etc
+            topicSuffixingStrategy = TopicSuffixingStrategy.SUFFIX_WITH_INDEX_VALUE,
+            // time to wait before attempting to retry
+            backoff = @Backoff(delay = 1000, multiplier = 2.0),
+            // if these exceptions occur, skip retry then push message to DLQ
+            exclude = {
+                    SerializationException.class,
+                    DeserializationException.class,
+                    DuplicateHL7FileFoundException.class,
+                    DiHL7Exception.class,
+                    HL7Exception.class,
+                    XmlConversionException.class,
+                    JAXBException.class
+            }
+
+    )
+    @KafkaListener(
+            topics = "${kafka.raw.xml-topic}"
+    )
+    public void handleMessageForElrXml(String message,
+                                       @Header(KafkaHeaders.RECEIVED_KEY) String messageId,
+                                       @Header(KafkaHeaders.RECEIVED_TOPIC) String topic) {
+        log.debug(topicDebugLog, messageId, topic);
+        NbsInterfaceModel nbsInterfaceModel = nbsRepositoryServiceProvider.saveElrXmlMessage(messageId, message);
```

Comment:
this is related to Version Header, as for this flow, use version value as indicate check,
if value is 2, we need to save data to RTI with different status (RTI_QUEUED) and send out another topic for RTI event

Reply:
https://github.com/CDCgov/NEDSS-DataIngestion/blob/main/data-ingestion-service/src/main/java/gov/cdc/dataingestion/kafka/integration/service/KafkaConsumerService.java

example how we do the RTI in regular ELR flow.

line 473, check the dataProcessingApplied variable, seeing how we save the status differently
line 502 and 504 are where we send event to different path; the else probably is no longer needed, but the if is crucial for RTI